### PR TITLE
Use correct api key type when checking if a service is over daily limit

### PR DESCRIPTION
**Pull request by smcmurtry**
# Summary | Résumé

We were hard-coding the api key type to be `KEY_TYPE_NORMAL`, which caused a bug where "test" api key sending caused an error in celery if the service was over it's daily limit. 

I have now tested this on both the `/email` and `/sms` endpoints and I am no longer seeing any celery errors when sending with a Test API key after the service has reached its daily limit.

### CODE REVIEW

File: app/celery/tasks.py [hunk 1]
```diff
@@ -412,9 +412,9 @@ def save_emails(self, _service_id: Optional[str], signed_notifications: List[Sig
         # service is whatever it was set to last in the for loop above.
         # at this point in the code we have a list of notifications (saved_notifications)
         # which could be from multiple services
-        check_service_over_daily_message_limit(KEY_TYPE_NORMAL, service)
         research_mode = service.research_mode  # type: ignore
         for notification_obj in saved_notifications:
+            check_service_over_daily_message_limit(notification_obj.key_type, service)
```
Comment (jimleroyer) [inline app/celery/tasks.py:417]: Should we have a default to `KEY_TYPE_NORMAL` just in case? Can we guarantee this will always be there?

Comment (smcmurtry) [reply app/celery/tasks.py:417]: We are defaulting to `KEY_TYPE_NORMAL` here: https://github.com/cds-snc/notification-api/blob/bug/test-key-sending-blocked/app/celery/tasks.py#L366

So I think the `key_type` will always be there.

**Review by andrewleith — COMMENTED** [review]
The key type is also hard coded in the `save_smss` function here: https://github.com/cds-snc/notification-api/blob/main/app/celery/tasks.py#L298

**Comment by smcmurtry** [comment]
Good catch @andrewleith! I even left a comment above that line a while ago. I will move that inside the for loop.

File: app/celery/tasks.py (hunk 1 above)
Comment (jimleroyer) [reply app/celery/tasks.py:416]: Thanks @smcmurtry . I very much like the new SignedNotification and VerifiedNotification types btw.

**Review by andrewleith — APPROVED** [review]
LGTM

**Comment by smcmurtry** [comment]
Oh I actually included these changes in this (now merged) PR since it was touching the same lines: https://github.com/cds-snc/notification-api/pull/1744

So this PR can be closed since the fix has been made. (I probably should have merged this 1st to make the changes easier to track)

**Pull request closed without merging**